File: 02-backend/spring-boot-ecommerce/src/main/java/com/td005/ecommerce/dto/Purchase.java
package com.td005.ecommerce.dto;

import com.td005.ecommerce.entity.Address;
import com.td005.ecommerce.entity.Customer;
import com.td005.ecommerce.entity.Order;
import com.td005.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;


}

File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
  customer:Customer|any;
  shippingAddress:Address|any;
  billingAddress:Address|any;
  order:Order|any;
  orderItems:OrderItem[]|any;


}
